internal/engine/synclet_build_and_deployer.go

```diff
@@ -130,15 +131,13 @@ func (sbd *SyncletBuildAndDeployer) updateInCluster(ctx context.Context, iTarget
 	containerPathsToRm := build.PathMappingsToContainerPaths(toRemove)
 
 	// archive files to copy to container
-	ab := build.NewArchiveBuilder(ignore.CreateBuildContextFilter(iTarget))
+	// TODO(dmiller) maybe don't do this?
```

Comment:
would this just be a pretty straightforward application of the pipe/go used elsewhere in this PR?

Reply:
The problem is both `updateViaExec` and `updateViaSynclet` take the paths that we archived as a parmater which the current API doesn't expose to us before the archiving is done, and now the archiving is done asynchronously in a separate goroutine.

Reply:
ah...It looks like `updateViaExec` only uses it for logging purposes. Maybe we can drop that param and then add a `-v` to the `tar xf` [here](https://github.com/windmilleng/tilt/blob/c0a9ce2d6b88f5f79a32aeac9ca50d81ce6332f3/internal/engine/synclet_build_and_deployer.go#L226).

And then IIUC, `updateViaSynclet` only takes the list of files to delete, not the list of files that we archived, so that's not a problem.

Reply:
Yeah I'm down for that. Let me fix these other test failure first.

Reply:
I'm also tempted not to fix this since it seems like the synclet may be getting deprecated?